kernel/src/processbuffer.rs

```diff
@@ -519,81 +456,155 @@ impl ReadableProcessBuffer for ReadWriteProcessBuffer {
         }
     }
 
-    fn enter<F, R>(&self, fun: F) -> Result<R, process::Error>
+    fn enter<'a, F, R>(&'a self, fun: F) -> Result<R, process::Error>
     where
-        F: FnOnce(&ReadableProcessSlice) -> R,
+        F: FnOnce(ReadableProcessSlice<'a>) -> R,
```

Comment:
This changes the lifetime behavior of `enter`. Previously the arg's lifetime was that of the function call, now it's the same as `&self`. If this is sound, and I'm not positive it is, we could instead be using a `fn get(&self) -> ReadableProcessSlice<'a>`. Considering it was switched to `enter` from a getter previously, what was the reasoning behind that?